packages/core/src/root.ts

```diff
@@ -360,6 +382,19 @@ export function createRemoteRoot<
           children: value.children.map((child) => serialize(child as any)),
         };
   }
+
+  function validateAllowedComponent(
+    component: CanBeChild,
+    allowedComponents: readonly AllowedComponents[],
+  ) {
+    const type = getComponentType(component);
+    const isAllowed = allowedComponents.indexOf(type as any) >= 0;
+    return isAllowed;
+  }
+
+  function getComponentType(component: CanBeChild) {
+    return component.kind === KIND_TEXT ? 'Text' : component.type;
```

Comment:
This is not safe because we have components with type `Text` also. IMO, true "text" nodes should always be allowed.